miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java

```diff
@@ -105,13 +105,32 @@ public void setDilutionService(LibraryDilutionService dilutionService) {
   }
 
   @RequestMapping(value = "{poolId}", method = RequestMethod.GET, produces = "application/json")
-  public @ResponseBody String getPoolById(@PathVariable Long poolId) throws IOException {
-    ObjectMapper mapper = new ObjectMapper();
+  public @ResponseBody PoolDto getPoolById(@PathVariable Long poolId) throws IOException {
     Pool p = poolService.getPoolById(poolId);
     if (p == null) {
       throw new RestException("No pool found with ID: " + poolId, Status.NOT_FOUND);
     }
-    return mapper.writeValueAsString(p);
+    return Dtos.asDto(p, true);
+  }
+
+  @RequestMapping(method = RequestMethod.POST, produces = "application/json")
+  @ResponseBody
+  public PoolDto createPool(@RequestBody PoolDto pool, UriComponentsBuilder uriBuilder, HttpServletResponse response)
+      throws IOException {
+    Long id = poolService.savePool(Dtos.to(pool));
+    UriComponents uriComponents = uriBuilder.path("/{id}").buildAndExpand(id);
```

Comment:
I believe it's relative to the current path. Now that we return the DTO, I think we can not bother.